manager/controllers/app/plotter_controller_test.go

```diff
@@ -0,0 +1,185 @@
+		})
+
+		// Plotters not deployed to m4d-system should not be successfully reconciled due to the filter preventing
+		// reconcile from being called.
+		It("Test Plotter Deploy to Bad Namespace", func() {
+			deployAndCheck("default", false)
+		})
+	})
+})
+
+var _ = Describe("Plotter Controller Legal Create Event", func() {
+	Context("Plotter", func() {
+		BeforeEach(func() {
+			// Add any setup steps that needs to be executed before each test
+		})
+
+		AfterEach(func() {
+			// Add any teardown steps that needs to be executed after each test
+		})
+
+		// Plotter are successfully reconciled when deployed to m4d-system only
+		It("Test Plotter Deploy to Correct Namespace", func() {
+			deployAndCheck(utils.GetSystemNamespace(), true)
+		})
+	})
+})
+
+// Create and then Delete the plotter and check that the result is what was expected.
+// Deploy the plotter and check that the result is what was expected.
+func deployDeleteAndCheck(namespace string, shouldSucceed bool) {
+	const timeout = time.Second * 30
+	const interval = time.Millisecond * 100
+
+	// Create the plotter yaml from the hard coded testdata file
+	plotter := &app.Plotter{}
+	Expect(readObjectFromFile("../../testdata/plotter.yaml", plotter)).ToNot(HaveOccurred())
+
+	// Set the namespace to the system namespace so that the plotter will be successfully created
+	plotter.SetNamespace(utils.GetSystemNamespace())
+
+	// Create the Plotter.  Need to wait in case it's still in the process of being deleted from a previous test.
+	By("Expecting the plotter to be created.")
+	Eventually(func() error {
+		return k8sClient.Create(context.Background(), plotter)
+	}, timeout, interval).Should(Succeed())
+
+	// Don't forget to clean up after tests finish
+	plotterKey := client.ObjectKeyFromObject(plotter)
+	defer func() {
+		plotter := &app.Plotter{ObjectMeta: metav1.ObjectMeta{Namespace: plotterKey.Namespace, Name: plotterKey.Name}}
+		_ = k8sClient.Get(context.Background(), plotterKey, plotter)
+		_ = k8sClient.Delete(context.Background(), plotter)
+	}()
+
+	By("Expecting to find the plotter we created")
+	Eventually(func() error {
+		return k8sClient.Get(context.Background(), plotterKey, plotter)
+	}, timeout, interval).Should(Succeed())
+
+	By("Expecting Plotter to eventually be ready")
+	Eventually(func() bool {
+		Expect(k8sClient.Get(context.Background(), plotterKey, plotter)).To(Succeed())
+		return plotter.Status.ObservedState.Ready
+	}, timeout, interval).Should(BeTrue(), "Plotter is not ready after timeout!")
+
+	// Now create a request to delete the same plotter
+	// Set the namespace to the namespace being used for this test
+	delPlotter := &app.Plotter{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: plotterKey.Name}}
+
+	// Delete the Plotter
+	if shouldSucceed {
+		Expect(k8sClient.Delete(context.Background(), delPlotter)).Should(Succeed())
+	} else {
+		Expect(k8sClient.Delete(context.Background(), delPlotter)).ShouldNot(Succeed())
```

Comment:
The event filter that is introduced does not prohibit creation, deletion etc events. It just won't call the reconcile loop from events that you filtered out. And as @shlomitk1 pointed out. This statement does not succeed because the plotter you try to delete does not exist and not because the event loop filters it out.

Reply:
@froesef Agreed.  Now that I think about it though, I don't think there is any way to test the delete filter if the create filter is in place.  There's no way to create the plotter outside of m4d-system and so there's no way to show that deleting it outside of m4d-system does not call the reconcile.  I propose to remove the Illegal Delete Event test completely.  Does this make sense to you?

Reply:
One could put counters into the controller and count how many times a reconcile loop was executed and check that it's still the same after deleting the plotter object. But that would mean modifying the code and adding a counter just so we can verify this one test. That's probably not the way to go. Let's drop this test.

Reply:
Agreed.  Removing